Xapo consolidation

jnewbery opened the pull request:
This is a contribution from @ajtowns at Xapo.

harding commented on _posts/en/2018-07-30-xapo-consolidation.md:
```diff
@@ -0,0 +1,79 @@
+---
+title: 'Field Report: Xapo UTXO Consolidation'
```
This is a good, neutral headline, but if you want to maybe make it a bit catchier, I'd suggest: "Field Report: Consolidation of 4 Million UTXOs at Xapo", or anything else that mentions the large extent of the effort.

moneyball commented:
+1 catchier

On Thu, Jul 19, 2018 at 2:56 PM David A. Harding <notifications@github.com>
wrote:

> *@harding* commented on this pull request.
> ------------------------------
>
> In _posts/en/2018-07-30-xapo-consolidation.md
> <https://github.com/bitcoinops/bitcoinops.github.io/pull/28#discussion_r203885186>
> :
>
> > @@ -0,0 +1,79 @@
> +---
> +title: 'Field Report: Xapo UTXO Consolidation'
>
> This is a good, neutral headline, but if you want to maybe make it a bit
> catchier, I'd suggest: "Field Report: Consolidation of 4 Million UTXOs at
> Xapo", or anything else that mentions the large extent of the effort.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/bitcoinops/bitcoinops.github.io/pull/28#pullrequestreview-138867006>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAT6A-IxPHNCgu26MwXQQrVq6zJMmentks5uIQBwgaJpZM4VXHZU>
> .
>

jnewbery commented:
We might consider adding a foreword, in order to attribute properly to aj and to let the reader know who's voice she's listening to. Something along the lines of "Bitcoin Optech member company Xapo recently did a large round of UTXO consolidations. AJ Towns, Xapo senior engineer, explains what they did....". I'm sure we can come up with better wording.

jnewbery commented:
We might consider adding a foreword, in order to attribute properly to aj and to let the reader know who's voice she's listening to. Something along the lines of "Bitcoin Optech member company Xapo recently did a large round of UTXO consolidations. AJ Towns, Xapo senior engineer, explains what they did....". I'm sure we can come up with better wording.

jnewbery commented:
Pasting some feedback from Steve and response from aj here:

>  I think the post is great as is, but one possible improvement would be to link to another source (if there is one?) explaining consolidation in more technical detail, or, providing some type of visualization as to what is happening with consolidation. I don't think you need to waste your time on this though if it is much work or you feel the marginal value is small.

and response from aj:

> I'm not sure if there is a particularly good reference for UTXO consolidation; there are:
>
> https://en.bitcoin.it/wiki/Techniques_to_reduce_transaction_fees#Consolidation (linked from newsletters #0, #2 and #3 already)
> https://en.bitcoin.it/wiki/How_to_cheaply_consolidate_coins_to_reduce_miner_fees (linked from above)
> https://bitgo.freshdesk.com/support/solutions/articles/27000044185-what-are-some-best-practices-regarding-the-usage-of-consolidations-and-fanouts- (not really any more detailed)
>
> oxt.me/entity/Xapo gives a bit of a visualisation based on estimates from public data, but it doesn't seem to provide a permalink or an easy graph to snapshot, you have to click on UTXOs tab and navigate your way to see the UTXO decline over the year to date. Perhaps a better visualisation is statoshi's, though it does all UTXOs not just Xapo's: https://statoshi.info/dashboard/db/unspent-transaction-output-set?panelId=6&fullscreen&from=1514759562608&to=1532039707168

harding commented:
What's the desired timeline on publication?  Is this something we need for launch, or is it at least a few days out?

harding commented:
I made a quick [diff](https://github.com/harding/bitcoinops.github.io/commit/bfd142c869c39359aee1ed5fdb23712be11517ea) that,

- Revises the title to the one proposed above (which has a thumbs-up from AJ)
- Adds a byline for AJ, as a simple resolution to the feeback in https://github.com/bitcoinops/bitcoinops.github.io/pull/28#issuecomment-406439987 .  This uses the title he used for himself in http://www.erisian.com.au/wordpress/2018/05/25/buying-in-and-selling-out
- Adds an screenshot of the 2018 UTXO set from Statoshi to maybe make the article more attractive.  Disclosure: I locally edited the style on that page to tweak the font sizes; I didn't touch anything that would've affected the data
- Adds an Additional Resources section to the bottom with the three additional links AJ provided in the quoted text in https://github.com/bitcoinops/bitcoinops.github.io/pull/28#issuecomment-406440281

With the title change, the h1 overflowed to the next line, which looked ugly to me, so I changed h1s to be centered.  This changes the h1 on every page on the site.  It looks good to me, but at least someone else should just flip through a few pages to ensure it looks good to them too.

I didn't review the changes to the site index page.  I think this needs to be rebased first.  Otherwise, ACK on the text.

![2018-07-20-07_27_43_845263355](https://user-images.githubusercontent.com/61096/43000205-8904334e-8bee-11e8-93c0-73239ecef9b3.png)

jnewbery commented:
Thanks for the review and diff @harding . I'll review later.

> What's the desired timeline on publication? Is this something we need for launch, or is it at least a few days out?

We don't need it for launch. I think it'd be good to get this published in time for the July 30th newsletter.

jnewbery commented:
I've reviewed @harding's changes and they look good to me. I really like the graph and additional links.

I didn't like centered `h1` throughout the site (particularly on the newsletter index page and the workshops page), so I changed that in #44 , which is merged into master.

I've rebased this on master. It looks good to me - once I have ACKs I'll squash the fixup commits and update the first commit log to credit @ajtowns and @harding .

This should be merged simultaneously with the next newsletter.

jnewbery commented:
fixed travis failure and force-pushed

jnewbery commented:
Squashed and updated commit log.

This could do with a final review before publication. @jamesob ?

jamesob commented:
Here's a [suggested patch](https://gist.github.com/jamesob/8f9272eabeea73d2713e3f31ac314d91). The first change is a grammatical error (my only blocking issue) and the rest are style suggestions.

jnewbery commented:
I'll let @ajtowns have final say on style nits.

Also aj, can you confirm that you're ok to release this under the MIT license (the license wasn't explicitly mentioned in this repo when you sent me the article).

harding commented:
ACK b57b238855370f5f5a8585b2ec30f41c6511ad70

I'm fine with the first sentence either the way it is now or after @jamesob's suggested modification to make the auxiliary "has/have" agree with the plural "months".  I can explain in grammatical terms why it works both ways if you care, but the simple solution is to remove the auxiliary and intensify the switch in tenses by changing "made" to "makes", so it reads "...the past few months of low transaction fees makes it a great time..."  Considering that the sentence already ends with an exclamation point, intensification works well here IMO.

Re: next Tuesday's newsletter that's expected to include this article.  I'm basing it on this branch and I made one non-visible change to allow the newsletter and the independent post to share exactly the same content: I moved the text of this post to the `_includes/` directory and then just included it back in to the current page.  The only thing I tweaked was the subhead for the Additional Resources section, which I made possible to specify its depth via the `{% include %}` directive so that it can be an H2 on the independent post but probably an H3 in the newsletter.  Here's that commit in case you want to make it a part of this branch directly: https://github.com/harding/bitcoinops.github.io/commit/94bf69e287923a1689e88c6868c91c2c91dd56e0

jnewbery commented:
We want to publish this tomorrow morning, so I've force pushed some of James's changes. I've taken @harding's suggested wording for the first sentence ("...the past few months of low transaction fees **makes** it a great time..).

I've also added @harding's change to move the content into `_includes` so it can be included in multiple places. I'll rebase #47 on top of that.

These commits will be merged as part of #47.

jnewbery commented:
Merged in #47
